routers/api/v2/tokens.go

```diff
@@ -53,14 +46,51 @@ func (r *apiV2Router) CreateServiceToken(ctx *gin.Context) {
 		}
 	}
 
-	token, err := r.authorizer.CreateServiceToken(req.Owner, parsedURIs, req.ExpiresAt)
+	userToken := r.GetAuthToken(ctx)
+	userID, err := r.authorizer.GetUserIdFromToken(userToken)
+	if err != nil {
+		r.logger.Debug("could not extract id from auth token", zap.Error(err))
+		models.SendAPIError(ctx, http.StatusBadRequest, "could not extract user id from auth token")
+		return
+	}
+
+	tokenID := primitive.NewObjectID()
+	token, err := r.authorizer.CreateServiceToken(tokenID, parsedURIs, req.ExpiresAt)
 	if err != nil {
 		r.logger.Error("could not create JWT", zap.Error(err))
 		models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
 		return
 	}
 
+	_, err = r.tokenService.AddServiceToken(ctx, tokenID, userID.Hex(), token)
+	if err != nil {
+		r.logger.Error("could not create service token", zap.Error(err))
+		models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
+		return
+	}
+
 	ctx.JSON(http.StatusOK, serviceTokenRes{
 		Token: token,
 	})
 }
+
+// DELETE: /api/v2/tokens/service/:id
+// Response:
+// Headers:  Authorization <- token
+func (r *apiV2Router) DeleteServiceToken(ctx *gin.Context) {
+	tokenID := ctx.Param("id")
+	if len(tokenID) == 0 {
+		r.logger.Debug("token id must be provided in request")
+		models.SendAPIError(ctx, http.StatusBadRequest, "token id must be provided")
+		return
+	}
+
+	err := r.tokenService.DeleteServiceToken(ctx, tokenID)
```

Comment:
Service token invalidation is also not a feature that's specific to the v2 API controller, tokens should be invalidated through the `Authorizer`.